hexa/workspaces/schema/mutations.py

```diff
@@ -114,23 +118,60 @@ def resolve_create_workspace_member(_, info, **kwargs):
         workspace: Workspace = Workspace.objects.filter_for_user(request.user).get(
             slug=input["workspaceSlug"]
         )
-        user = User.objects.get(email=input["userEmail"])
-
-        workspace_membership = WorkspaceMembership.objects.create_if_has_perm(
-            principal=request.user, workspace=workspace, user=user, role=input["role"]
-        )
-        send_mail(
-            title=gettext_lazy(f"You've been added to the workspace {workspace.name}"),
-            template_name="workspaces/mails/invite_member",
-            template_variables={
-                "workspace": workspace.name,
-                "owner": request.user.display_name,
-                "workspace_url": request.build_absolute_uri(
-                    f"//{settings.NEW_FRONTEND_DOMAIN}/workspaces/{workspace.slug}"
+        _user = User.objects.filter(email=input["userEmail"])
+        params = {}
+        workspace_membership = None
+        if _user:
+            user = _user.first()
+            workspace_membership = WorkspaceMembership.objects.create_if_has_perm(
+                principal=request.user,
+                workspace=workspace,
+                user=user,
+                role=input["role"],
+            )
+            params = {
+                "title": gettext_lazy(
+                    f"You've been added to the workspace {workspace.name}"
                 ),
-            },
-            recipient_list=[user.email],
-        )
+                "template_name": "workspaces/mails/invite_member",
+                "template_variables": {
+                    "workspace": workspace.name,
+                    "owner": request.user.display_name,
+                    "workspace_url": request.build_absolute_uri(
+                        f"//{settings.NEW_FRONTEND_DOMAIN}/workspaces/{workspace.slug}"
+                    ),
+                },
+                "recipient_list": [user.email],
+            }
+        else:
+            invitation = WorkspaceInvitation.objects.create_if_has_perm(
+                principal=request.user,
+                workspace=workspace,
+                email=input["userEmail"],
+                role=input["role"],
+            )
+            signer = TimestampSigner()
+            token = base64.b64encode(
+                signer.sign(invitation.id).encode("utf-8")
+            ).decode()
+            user_email = parse.quote(input["userEmail"])
```

Comment:
Since you store It in your model, you can safely use the value without using `parse.quote`/